### PR TITLE
RUST-911: Add srvServiceName URI option

### PR DESCRIPTION
**Pull request by jadalilleboe**
[RUST-911](https://jira.mongodb.org/browse/RUST-911)

### CODE REVIEW

**Review by isabelatkinson — COMMENTED** [review]
looking good so far - we'll also need to implement the prose test described [here](https://github.com/mongodb/specifications/tree/master/source/polling-srv-records-for-mongos-discovery/tests#srv-polling-with-srvservicename-mongoclient-option)

**Comment by isabelatkinson** [comment]
It looks like we're also currently skipping some URI options tests for `srvServiceName` that we should unskip as part of this. See [here](https://github.com/mongodb/mongo-rust-driver/blob/842c536381107f6f181196cf38a36fc8da136ba5/src/client/options/test.rs#L25) and [here](https://github.com/mongodb/mongo-rust-driver/blob/842c536381107f6f181196cf38a36fc8da136ba5/src/test/spec/initial_dns_seedlist_discovery.rs#L67). (I think the comment on the second one points to the wrong ticket.)

File: src/sdam/srv_polling/test.rs [hunk 1]
```diff
@@ -182,7 +203,22 @@ async fn srv_max_hosts_random() {
...
     assert!(actual.contains(&localhost_test_build_10gen(27017)));
 }
+
+#[tokio::test]
+async fn srv_service_name() {
+    let hosts = vec![
+        localhost_test_build_10gen(27019),
+        localhost_test_build_10gen(27020),
+    ];
+    run_test_srv(
```
Comment (isabelatkinson) [inline src/sdam/srv_polling/test.rs:217]: `run_test_srv` just returns the addresses on the topology, so we'll need to add assertions on what gets returned here (see some of the other tests for examples)

**Comment by isabelatkinson** [comment]
disregard my comment about assertions, I misread which method was being called!

**Review by isabelatkinson — APPROVED** [review]
lgtm!

File: src/client/options.rs [hunk 2]
```diff
@@ -900,11 +910,16 @@ impl Default for HostInfo {
 }
 
 impl HostInfo {
-    async fn resolve(self, resolver_config: Option<ResolverConfig>) -> Result<ResolvedHostInfo> {
+    async fn resolve(
+        self,
+        resolver_config: Option<ResolverConfig>,
+        srv_service_name: Option<String>,
```
Comment (abr-egn) [inline src/client/options.rs:916]: Would it make sense to bundle `srv_service_name` into `resolver_config`?  Mechanically, they're always passed around as a pair but I could see going either way on whether it's the right thing conceptually.

**Review by abr-egn — APPROVED** [review]
LGTM, with one judgement call I leave up to you :)  No need for re-review if you decide to keep it as is.

File: src/client/options.rs (hunk 2 above)
Comment (jadalilleboe) [reply src/client/options.rs:916]: Looked into this a bit, decided to keep it as is to reduce complexity in how the srv_service_name is passed through the codepath (from being parsed as part of the connection string down to the srv resolver- adding the parameter to ResolverConfig disrupts this a little bit). Thank you for pointing this out!

**Pull request merged**